common/transform/src/main/java/ai/h2o/mojos/deploy/common/transform/MojoScorer.java

```diff
@@ -64,13 +87,103 @@ public MojoScorer(
    * @return response {@link ScoreResponse}
    */
   public ScoreResponse score(ScoreRequest request) {
-    MojoFrame requestFrame = requestConverter.apply(request, pipeline.getInputFrameBuilder());
+    MojoFrame requestFrame = scoreRequestConverter
+            .apply(request, pipeline.getInputFrameBuilder());
     MojoFrame responseFrame = doScore(requestFrame);
-    ScoreResponse response = responseConverter.apply(responseFrame, request);
+    ScoreResponse response = scoreResponseConverter.apply(responseFrame, request);
     response.id(pipeline.getUuid());
+    try {
+      ShapleyType requestedShapleyType = shapleyType(request.getShapleyValuesRequested());
+      switch (requestedShapleyType) {
+        case TRANSFORMED:
+          response.setFeatureShapleyContributions(computeContribution(request));
+          break;
+        case ORIGINAL:
+          log.info(UNIMPLEMENTED_MESSAGE);
+          break;
+        default:
+          break;
+      }
+    } catch (Exception e) {
+      log.info("Failed shapley values: {}, due to: {}", request, e.getMessage());
+      log.debug(" - failure cause: ", e);
+    }
```

Comment:
We should always when we cannot fulfill the request. "Failing silently" causes a lot of "why does it not return what I have requested" situations.

Reply:
This was reg a discussion about 

> As per discussion on slack, my above comment is related to:
> #165
> #62
> 
> Since not specific to shapley scores. Deferring to later changes. Discussed with @Rajimut that we can add additional field in response:
> message which is a map of {message_level: <enum of: LOG, WARN, ERROR>, message: <text of message>} or something.

The idea is to not fail the scoring request when shapley values cannot be computed for a model - And in mojos that are not obtained from DAI like h2o-3 mojos we out shapley requests will fail causing the scoring response to fail as well. In order to handle this, as per discussion above, we have decided to provide 400 response on shapley exclusive endpoint `/model/contribution`. The existing endpoint `/model/score` will remain unaffected if the shapley values are not available.

[For the future] We also want to include a message field with the score and shapley response to provide some additional information which will describe the reason for the failure

Reply:
Some thoughts:
- If a user explicitly requests Shap scores for a scorer that has loaded an H2O-3 MOJO (perhaps the client even knowing that the feature isn't supported), would we consider that to be client error?  Should it be the server's responsibility to "fix/adjust" a client's request?
- The try-catch block is catching all shap errors.  If the error is due to something other than compatibility, silently failing would hide all of it.
- I envision the two points above on the same level as how we fail if one of the regular client-provided scoring row is malformed.  We don't currently ignore one row while allowing the rest to continue.